### ISSUE TITLE
Have you achieved the project?

### ISSUE THREAD

**Issue by calmelo** [rmukhia/pegasus#2]
Can the project be used? I want to try.

**Comment by rmukhia** [comment]
Hi, I did this project for my Master's Thesis and I was successfully able too control 1 drone in real life and 3 simultaneous drones in simulation. 

The results and conclusion can be found here.
<https://github.com/rmukhia/AIT-Thesis/blob/master/main.pdf>

**Comment by calmelo** [comment]
> Hi, I did this project for my Master's Thesis and I was successfully able too control 1 drone in real life and 3 simultaneous drones in simulation.
> 
> The results and conclusion can be found here. https://github.com/rmukhia/AIT-Thesis/blob/master/main.pdf

Are the codes released here?

**Comment by calmelo** [comment]
> Hi, I did this project for my Master's Thesis and I was successfully able too control 1 drone in real life and 3 simultaneous drones in simulation.
> 
> The results and conclusion can be found here. https://github.com/rmukhia/AIT-Thesis/blob/master/main.pdf

How can I use this codes?

**Comment by rmukhia** [comment]
can i know your intent to use it? 
I have spent a lot of time and effort to finish this project. and from my masters thesis pdf I have achieved the objectives defined as described in my thesis report. 
the code is usable, although not state of the art. 
please define your objective to me before running my code.
my goal is to have an automous multi-drone system that can solve the muti agent complete coverage problem. I will be happy to collaborate and extend my thesis academically...

**Comment by calmelo** [comment]
> can i know your intent to use it? I have spent a lot of time and effort to finish this project. and from my masters thesis pdf I have achieved the objectives defined as described in my thesis report. the code is usable, although not state of the art. please define your objective to me before running my code. my goal is to have an automous multi-drone system that can solve the muti agent complete coverage problem. I will be happy to collaborate and extend my thesis academically...

I want to simulate the multi-drone system using px4 and ns3. I have achieved simulate the multi-drone system using px4, but I can't integrate the ns3 into the system. This is my research, so I want to reappearance the project and learn.

**Comment by calmelo** [comment]
> can i know your intent to use it? I have spent a lot of time and effort to finish this project. and from my masters thesis pdf I have achieved the objectives defined as described in my thesis report. the code is usable, although not state of the art. please define your objective to me before running my code. my goal is to have an automous multi-drone system that can solve the muti agent complete coverage problem. I will be happy to collaborate and extend my thesis academically...

How can I run this codes? Maybe I only need the px4&gazebo simulator to communicate with the ns3, so I don't need something like path planning and gscam. However, your paper don't describe that how to run this codes.

**Comment by calmelo** [comment]
> can i know your intent to use it? I have spent a lot of time and effort to finish this project. and from my masters thesis pdf I have achieved the objectives defined as described in my thesis report. the code is usable, although not state of the art. please define your objective to me before running my code. my goal is to have an automous multi-drone system that can solve the muti agent complete coverage problem. I will be happy to collaborate and extend my thesis academically...

Can you write a instruction for using the codes? I will be very appreciate it. I'm looking forward to your reply! Thanks!

**Comment by rmukhia** [comment]
Sorry I tried to run the project during weekend but my current OS/tools does not support the this project unless I reinstall the older OS and requirements. 

However what you need to do is use Gazebo client library and include in your NS3 application. And build the NS3 application. The application should be both a Gazebo client as well as ns3 application. You can use the output from Gazebo simulation as the position for moving the nodes in ns3. 

pegasus-net-sim is a ns3 application and you can compile it using the standard way you compile ns3 application which you can find plenty of documentation about. And you need to install gazebo development headers and libraries.


> I want to simulate the multi-drone system using px4 and ns3. I have achieved simulate the multi-drone system using px4, but I can't integrate the ns3 into the system. This is my research, so I want to reappearance the project and learn.
This is my research too. And I want to extend it in the future. Can you please tell me your university because it seems like your university is pursuing research in similar field and I would like to apply. :D

**Comment by calmelo** [comment]
> Sorry I tried to run the project during weekend but my current OS/tools does not support the this project unless I reinstall the older OS and requirements.
> 
> However what you need to do is use Gazebo client library and include in your NS3 application. And build the NS3 application. The application should be both a Gazebo client as well as ns3 application. You can use the output from Gazebo simulation as the position for moving the nodes in ns3.
> 
> pegasus-net-sim is a ns3 application and you can compile it using the standard way you compile ns3 application which you can find plenty of documentation about. And you need to install gazebo development headers and libraries.
> 
> > I want to simulate the multi-drone system using px4 and ns3. I have achieved simulate the multi-drone system using px4, but I can't integrate the ns3 into the system. This is my research, so I want to reappearance the project and learn.
> 
> This is my research too. And I want to extend it in the future. Can you please tell me your university because it seems like your university is pursuing research in similar field and I would like to apply. :D

I install gazebo, and copy the pegasus-net-sim into the scratch folder in the ns3. However, when build the NS3 application, there is errors, and it cannot find gazebo. Do you know how to solve the problem?
![image](https://github.com/rmukhia/pegasus/assets/61192104/15534c5f-5992-49a8-b598-6ce66fbcc475)

Wuhan University.

**Comment by rmukhia** [comment]
Humm, it looks like the code needs to be updated. I ran it on Ubuntu 18 (ROS melodic), with Gazebo 9 which was available at around 2020. Your versions might be different. 
From your error it seems 'transport.hh' file does not exist in newer gazebo anymore or you have not installed gazebo development headers properly.

**Comment by calmelo** [comment]
> Humm, it looks like the code needs to be updated. I ran it on Ubuntu 18 (ROS melodic), with Gazebo 9 which was available at around 2020. Your versions might be different. From your error it seems 'transport.hh' file does not exist in newer gazebo anymore or you have not installed gazebo development headers properly.

No, I can find the the file in /usr. However, the ns3 cannot find it, and I don't know how to add the library, and I know how to add it in cmakelist. Do you know how to add the library in wscript?
My environment is also Ubuntu 18 (ROS melodic), with Gazebo 9.
If the code needs to be updated, could you provide me the codes which you ran successfully? I will be very appreciated for it!
Something importantly, can you write a document to teach us to use your project? Your project is very good, and I really want to reproduce it.
Thanks for your reply!

**Comment by calmelo** [comment]
> Humm, it looks like the code needs to be updated. I ran it on Ubuntu 18 (ROS melodic), with Gazebo 9 which was available at around 2020. Your versions might be different. From your error it seems 'transport.hh' file does not exist in newer gazebo anymore or you have not installed gazebo development headers properly.

Could you share me the right code, which can run correctly?

**Comment by rmukhia** [comment]
will you give me credit for the project.. also, if you publish paper, will you put me as an author. this is not a published thesis, I have put it on hold right now because I am doing other work right now. can I trust you?
